### PR TITLE
feat(router): Create a merchant config for enable processor agnostic MIT

### PR DESCRIPTION
**Pull request by ShankarSinghC**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

 This pr adds a merchant config (`pg_agnostic_mit`) to enable usage of Network Transaction/Reference ID in recurring `MIT` payments. This will be a `business profile` level config that a merchant specifies.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Set processor agnostic MIT config as `true`
image
config status from db
image
Set processor agnostic MIT config as `false`
image
config status from db
image

```
curl --location 'http://localhost:8080/routing/business_profile/{business_profile_id}/configs/pg_agnostic_mit' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: api-key' \
--data '{
    "enabled": true
}'
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/routing.rs [hunk 1]
```diff
@@ -818,6 +817,37 @@ pub async fn retrieve_linked_routing_config(
     }
 }
 
+pub async fn create_connector_agnostic_mandate_config(
```
Comment (vspecky) [inline crates/router/src/core/routing.rs:820]: Make this an `upsert` logic instead.
```rust
let config = state.find_config();
match config {
    Ok(conf) => { update },
    Err(err) if err.is_db_not_found() => { insert },
    Err(err) => { throw error }
}
```

Comment (vspecky) [reply crates/router/src/core/routing.rs:820]: additionally, add logic to the "create business profile" core function to insert a default value to the configs table from the start.

**Review by vspecky — CHANGES_REQUESTED** [review]

File: crates/router/src/core/routing.rs (hunk 1 above)
Comment (ShankarSinghC) [reply crates/router/src/core/routing.rs:820]: This function `find_config_by_key_unwrap_or` has the same logic.

Comment (Aprabhat19) [inline crates/router/src/core/routing.rs:820]: Nit:
```suggestion
pub async fn create_connector_agnostic_mandate_config(
```
pub async fn upsert_connector_agnostic_mandate_config(

**Review by Aprabhat19 — APPROVED** [review]

**Review by Aprabhat19 — APPROVED** [review]

**Review by Chethan-rao — APPROVED** [review]

File: crates/router/src/routes/app.rs [hunk 2]
```diff
@@ -437,6 +437,10 @@ impl Routing {
                         )
                     })),
             )
+            .service(
+                web::resource("/business_profile/{business_profile_id}/configs/pg_agnostic_mit")
```
Comment (Narayanbhat166) [inline crates/router/src/routes/app.rs:441]: this endpoint should not have business profile in the path params

**Pull request merged**